app/controllers/signup.controller.js

```diff
@@ -0,0 +1,130 @@
+const { updateValidator, createValidator, retrieveValidator } = require("../../validators/signup");
+const Signup = require("../models/signup.model.js");
+
+//create a Product
+exports.createSignup = (req, res) => {
+    const { error, value } = createValidator.validate(req.body)
+    if (error) {
+        return res.status(400).send(error.details)
+    }
+    signup = new Signup(value);
+
+    // Save Product in database
+    signup.save()
+        .then((data) => {
+            return res.send(data);
+        })
+        .catch((err) => {
+            return res.status(500).send({
+                message: err.message || "error occured while creating the product.",
+            });
+        });
+};
+
+//Retrieve and return all product from database.
+exports.findAllSignup = (req, res) => {
+    Signup.find()
+        .then((signup) => {
+            res.send(signup);
+        })
+        .catch((err) => {
+            res.status(500).send({
+                message: err.message || "error occured while gettin products.",
+            });
+        });
+};
+
+//Find a single products with id
+exports.findOneSignup = (req, res) => {
+    if (error) {
+        return res.status(400).send(error.details)
+    }
+    Signup.findById(req.params.id)
+        .then((signup) => {
+            if (!signup) {
+                return res.status(404).send({
+                    message: "not found with id" + req.params.id,
+                });
+            }
+            res.send(signup)
+        })
+        .catch((err) => {
+            if (err.kind === "Objectid") {
+                return res.status(404).send({
+                    message: "not found with id" + req.params.id,
+                });
+            }
+            return res.status(500).send({
+                message: "Error finding with id" + req.params.id,
+            });
+        });
+};
+
+// Updated a product identified by the id in the request
+exports.updateSignup = (req, res) => {
+    if (error) {
+        return res.status(400).send(error.details)
+    }
+    if (!req.body.name) {
+        return res.status(400).send({
+            message: "names can not be empty",
+        });
+    }
+    // Find product and update it with the request body
```

Comment:
where is Product coming from. the comment is saying another thing from the function name